### ISSUE TITLE
The "Resolve Unknown Type" action was not showing up when hovering over the class name.

### ISSUE THREAD

**Issue by loralei28** [microsoft/vscode-maven#932]
**Environment**

Operating System: Windows/Mac
JDK version: 17.0.3
Visual Studio Code version: 1.74.3
Java extension version: 1.14.0

**Steps To Reproduce**

1. Open a maven project
2. Open a .java file. In body of a function, type below statements.

```
Gson gson;
StringUtils.capitalize("abc");
```

**Current Result**
Windows
![image](https://user-images.githubusercontent.com/108506566/212857190-5ded072d-8c18-4568-9522-248cc692db63.png)
Mac
![image](https://user-images.githubusercontent.com/108506566/212858840-79b303f4-ec43-45f7-9cbb-3f188afd65e1.png)

**Expected Result**
![image](https://user-images.githubusercontent.com/108506566/212857168-d14cbdcc-fde1-4cdc-9cde-426767335daf.png)

**Comment by Eskibear** [comment]
- Works well with redhat.java@1.3.0
- "No delegate command" error with redhat.java@1.4.0.

**Comment by Eskibear** [comment]
```
org.osgi.framework.BundleException: Could not resolve module: com.microsoft.java.maven.plugin [164]
  Unresolved requirement: Require-Bundle: org.slf4j.api

	at org.eclipse.osgi.container.Module.start(Module.java:463)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel$2.run(ModuleContainer.java:1852)
	at org.eclipse.osgi.internal.framework.EquinoxContainerAdaptor$1$1.execute(EquinoxContainerAdaptor.java:136)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1845)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1786)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.doContainerStartLevel(ModuleContainer.java:1750)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1672)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:234)
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:345)
```

**Comment by Eskibear** [comment]
follow up actions:
- [ ] investigate why m2e removes slf4j.api, in case there's security issue.
- [ ] update dependencies for potential vulnerabilities

**Comment by Eskibear** [comment]
Bundle name for slf4j.api has changed since 1.14.0

```
% ls ~/.vscode/extensions/redhat.java-1.13.0-darwin-arm64/server/plugins/ | grep slf4j.api
org.slf4j.api_1.7.30.v20200204-2150.jar

% ls ~/.vscode/extensions/redhat.java-1.14.0-darwin-arm64/server/plugins/ | grep slf4j.api
slf4j.api_1.7.36.jar
```

Ideally we can simply change the bundle name accordingly, but that would break users who are still using redhat.java@1.13.0. I prefer we stay at current workaround per #933.

**Comment by Eskibear** [comment]
The fix has been shipped with pre-release v0.40.2023011807. @loralei28 you can switch to pre-release channel to give it a try.
